### PR TITLE
Convert irc package to js

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core

### CODE REVIEW

**Comment by MartinSchoeler** [comment]
@ggazzo i will need your help testing this

File: packages/rocketchat-irc/server/server.js [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+	}
+
+	onConnect() {
+		console.log('[irc] onConnect -> '.yellow, this.user.username, 'connect success.');
+		this.socket.write(`NICK ${ this.user.username }\r\n`);
+		this.socket.write(`USER ${ this.user.username } 0 * :${ this.user.name }\r\n`);
+		// message order could not make sure here
+		this.isConnected = true;
+		const messageBuf = this.msgBuf;
+		messageBuf.forEach(msg => {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: one line

File: packages/rocketchat-irc/server/server.js [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+	}
+
+	onQuitMember(member) {
+		console.log('[irc] onQuitMember ->'.yellow, 'username:', member);
+		RocketChat.models.Rooms.removeUsernameFromAll(member);
+		Meteor.users.update({ name: member }, { $set: { status: 'offline' }});
+	}
+
+	createUserWhenNotExist(name) {
+		let user;
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: this logic is seem good, but at least use one line to declare and set the variable.

File: packages/rocketchat-irc/server/server.js [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+		return { t: 'd', _id: rid };
+	}
+}
+
+IrcClient.getByUid = function(uid) {
+	return ircClientMap[uid];
+};
+
+IrcClient.create = function(login) {
+	let ircClient;
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: declare where the variable is used. (line 389).

File: packages/rocketchat-irc/server/server.js [hunk 4]
```diff
@@ -0,0 +1,455 @@
+//////
+// Assign values
+
+//Package availability
+const IRC_AVAILABILITY = RocketChat.settings.get('IRC_Enabled');
+
+// Cache prep
+const net = Npm.require('net');
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: I think that you should use `import`

**Comment by MartinSchoeler** [comment]
@ggazzo fixed the things you mentioned

File: packages/rocketchat-irc/server/server.js [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+		return login;
+	}
+	if (!(login.user._id in ircClientMap)) {
+		const ircClient = new IrcClient(login);
+		return async(ircClient.connect);
+	}
+	return login;
+};
+
+class IrcLoginer {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: use a function instead class

File: packages/rocketchat-irc/server/server.js [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+	constructor(login) {
+		console.log('[irc] validateLogin -> '.yellow, login);
+		return IrcClient.create(login);
+	}
+}
+
+class IrcSender {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: use a function instead class

File: packages/rocketchat-irc/server/server.js [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+		}
+		const room = RocketChat.models.Rooms.findOneById(message.rid, { fields: { name: 1, usernames: 1, t: 1 }});
+		const ircClient = IrcClient.getByUid(message.u._id);
+		ircClient.sendMessage(room, message);
+		return message;
+	}
+
+}
+
+class IrcRoomJoiner {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: use a function instead class

File: packages/rocketchat-irc/server/server.js [hunk 8]
```diff
@@ -0,0 +1,452 @@
...
+	constructor(user, room) {
+		const ircClient = IrcClient.getByUid(user._id);
+		ircClient.joinRoom(room);
+		return room;
+	}
+
+}
+
+class IrcRoomLeaver {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: use a function instead class

File: packages/rocketchat-irc/server/server.js [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+	constructor(user, room) {
+		const ircClient = IrcClient.getByUid(user._id);
+		ircClient.leaveRoom(room);
+		return room;
+	}
+
+}
+
+class IrcLogoutCleanUper {
```
Comment (ggazzo) [inline packages/rocketchat-irc/server/server.js]: use a function instead class

**Comment by ggazzo** [comment]
@haosdent @toxicgumbo could you help us to verify if this package still works?

**Pull request merged**